assets/test/e2e/test.js

```diff
@@ -147,22 +140,80 @@ describe('', function() {
+    describe('when adding to project', function() {
+      it('should view the create page', function() {
+        browser.get('/project/test/create');
+        expect(element(by.css('h1')).getText()).toEqual("Create Using Your Code");
+        expect(element(by.css('h3 a')).getText()).toContain("ruby-helloworld-sample");
+      });
+
+      it('should create from source', function() {
+        var sourceUrl = "https://github.com/openshift/rails-ex";
+        var appName = "my-rails-ex";
+
+        browser.get('/project/test/create');
+        element(by.name('from_source_url')).sendKeys(sourceUrl);
+        var nextButton = element(by.buttonText("Next"));
+        browser.wait(protractor.ExpectedConditions.elementToBeClickable(nextButton), 3000);
+
+        nextButton.click().then(function() {
+          expect(browser.getCurrentUrl()).toMatch(/\/project\/test\/catalog\/images/);
+          expect(element(by.css('h1')).getText()).toEqual("Select a builder image");
+          var builderImageLink = element(by.cssContainingText('a', "ruby"));
+          expect(builderImageLink.isPresent()).toBe(true);
+
+          builderImageLink.click().then(function() {
+            expect(browser.getCurrentUrl()).toMatch(/\/project\/test\/create\/fromimage/);
+            expect(element(by.css('h1')).getText()).toEqual("ruby");
+            element(by.name('appname')).sendKeys(appName);
+
+            element(by.buttonText("Create")).click().then(function() {
+              expect(browser.getCurrentUrl()).toMatch(/\/project\/test\/overview/);
+              expect(element(by.css('h3')).getText()).toContain("Created application " + appName + " in project test");
+            });
+          });
+        });
+      });
+
+      // TODO: create from template flow
+    });
+
+    describe('when creating a new project', function() {
+      it('should be able to show the create project page', function() {
+        browser.get('/createProject');
+        expect(element(by.cssContainingText("h1","New Project")).isPresent()).toBe(true);
+      });
+
+      it('should successfully create a new project', function() {
+        var project = {
+          name:        'e2e-test-project',
+          displayName: 'End-to-end Test Project',
+          description: 'Project created by the e2e web console tests.'
+        };
+
+        browser.get('/createProject');
+        for (var key in project) {
+          element(by.model(key)).sendKeys(project[key]);
+        }
+        element(by.buttonText("Create")).click();
+        expect(browser.getCurrentUrl()).toContain("/project/" + project['name'] + "/overview");
+        expect(element(by.cssContainingText("h1", project['displayName'])).isPresent()).toBe(true);
+
+        // all fields must have been honored
+        browser.get('/project/' + project['name'] + '/settings');
+        expect(element.all(by.css("dd")).get(0).getText()).toEqual(project['name']);
```

Comment:
Yes, I'm on it. I'll probably add some more markup to make the selector very explicit about what we want.